common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -6001,18 +6006,24 @@ public void verifyAndSet(String name, String value) throws IllegalArgumentExcept
       throw new IllegalArgumentException("Cannot modify " + name + " at runtime. It is in the list"
           + " of parameters that can't be modified at runtime or is prefixed by a restricted variable");
     }
-    String oldValue = name != null ? get(name) : null;
-    if (name == null || value == null || !value.equals(oldValue)) {
-      // When either name or value is null, the set method below will fail,
-      // and throw IllegalArgumentException
-      set(name, value);
+    if (!isLockedConfig(name)) {
```

Comment:
I prefer early return because it could be possible to add another restriction in the future.

Reply:
can you elaborate more on this?

Reply:
I guess it is more flexible if we return here.

```
if (isLockedConfig(name)) {
  return;
}
String oldValue = name != null ? get(name) : null;
...
```

Because (1) a kind of cognitive load is lower when we read the succeeding lines, (2) we can easily reorder the priority of the whitelist vs restrict list vs locked list, and (3) we can easily add another new condition like below without an additional nest.

```
if (isLockedConfig(name)) {
  return;
}
if (isProhibitedKeyAndValue(name, value)) {
  return;
}
```

I personally don't prefer the following shape.

```
if (!isLockedConfig(name)) {
  if (!isProhibitedKeyAndValue(name, value)) {
    String oldValue = name != null ? get(name) : null;
    ...
  }
}
```

Reply:
Understood